addons/binding/org.openhab.binding.plclogo/src/main/java/org/openhab/binding/plclogo/handler/PLCDateTimeHandler.java

```diff
@@ -0,0 +1,270 @@
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.builder.ChannelBuilder;
+import org.eclipse.smarthome.core.thing.binding.builder.ThingBuilder;
+import org.eclipse.smarthome.core.thing.type.ChannelTypeUID;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.plclogo.internal.PLCLogoClient;
+import org.openhab.binding.plclogo.internal.config.PLCDateTimeConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.ImmutableMap;
+
+import Moka7.S7;
+import Moka7.S7Client;
+
+/**
+ * The {@link PLCDateTimeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Alexander Falkenstern - Initial contribution
+ */
+@NonNullByDefault
+public class PLCDateTimeHandler extends PLCCommonHandler {
+
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Collections.singleton(THING_TYPE_DATETIME);
+
+    private final Logger logger = LoggerFactory.getLogger(PLCDateTimeHandler.class);
+    private PLCDateTimeConfiguration config = getConfigAs(PLCDateTimeConfiguration.class);
+
+    /**
+     * Constructor.
+     */
+    public PLCDateTimeHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (!isThingOnline()) {
+            return;
+        }
+
+        final PLCLogoClient client = getLogoClient();
+        final Channel channel = thing.getChannel(channelUID.getId());
+        if (!isValid(config.getBlockName()) || (channel == null)) {
+            logger.warn("Can not update channel {}: {}.", channelUID, client);
+            return;
+        }
+
+        final int address = getAddress(config.getBlockName());
+        if (address != INVALID) {
+            if (command instanceof RefreshType) {
+                final byte[] buffer = new byte[getBufferLength()];
+                int result = client.readDBArea(1, 0, buffer.length, S7Client.S7WLByte, buffer);
+                if (result == 0) {
+                    updateChannel(channel, S7.GetShortAt(buffer, address));
+                } else {
+                    logger.warn("Can not read data from LOGO!: {}.", S7Client.ErrorText(result));
+                }
+            } else if (command instanceof DateTimeType) {
+                final byte[] buffer = new byte[2];
+                final String type = channel.getAcceptedItemType();
+                if ((type != null) && type.equalsIgnoreCase(DATE_TIME_ITEM)) {
```

Comment:
Position literals first in `String` comparisons - that way if the `String` is `null` you won't get a `NullPointerException`, it'll just return `false`. [source: PMD](http://pmd.sourceforge.net/pmd-4.3.0/rules/design.html#PositionLiteralsFirstInComparisons)